### PR TITLE
Deposit Merkle Tree Implementation

### PR DESCRIPTION
**Pull request by dangerousfood**
Implementation of the Merkle tree for deposits

### CODE REVIEW

File: services/powchain/src/main/java/tech/pegasys/artemis/services/powchain/PowchainService.java [hunk 1]
```diff
@@ -136,7 +139,25 @@ public void run() {
...
+                              UnsignedLong.ZERO,
+                              depositInput);
+
+                      eventBus.post(
+                          new tech.pegasys.artemis.datastructures.operations.Deposit(
+                              null,
+                              UnsignedLong.valueOf(
+                                  deposit.getMerkle_tree_index().toLong(ByteOrder.LITTLE_ENDIAN)),
+                              deposit_data));
+                      // eventBus.post(deposit);
```
Comment (jrhea) [inline services/powchain/src/main/java/tech/pegasys/artemis/services/powchain/PowchainService.java]: is this supposed to be commented out?

File: ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/StateProcessor.java [hunk 2]
```diff
@@ -107,9 +115,15 @@ public void onEth2GenesisEvent(Eth2GenesisEvent event) {
...
-    LOG.log(Level.INFO, "Deposit Event detected: " + event.toString());
+  public void onDeposit(Deposit deposit) {
+    if (deposits == null) deposits = new ArrayList<Deposit>();
+    deposits.add(deposit);
   }
+  //
+  //  @Subscribe
+  //  public void onDepositEvent(DepositEvent event) {
+  //    LOG.log(Level.INFO, "Deposit Event detected: " + event.toString());
+  //  }
```
Comment (jrhea) [inline ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/StateProcessor.java:122]: can we delete this commented out code?

**Review by jrhea — APPROVED** [review]
lgtm

File: services/powchain/src/main/java/tech/pegasys/artemis/services/powchain/PowchainService.java (hunk 1 above)
Comment (dangerousfood) [reply services/powchain/src/main/java/tech/pegasys/artemis/services/powchain/PowchainService.java]: Removed in latest push

**Pull request merged**